### PR TITLE
Apply Clang-Tidy's suggestions

### PR DESCRIPTION
**Pull request by ice1000**
Signed-off-by: ice1000 <ice1000kotlin@foxmail.com>

Applied some suggestions from Clang-Tidy to grpc_wrap.cc since it's originally C but now C++

### CODE REVIEW

**Comment by overvenus** [comment]
Nice! Could you create a script to run clang tidy, like #209 ? Also we should use the same config as grpc's https://github.com/pingcap/grpc/blob/1de249c8a21563167152b1c7b794f9f4e4dcb7d0/.clang-tidy

**Comment by ice1000** [comment]
No, Clang-Tidy only provides suggestions, it's CLion that provide fixes. These changes are done in my IDE.
IDE also gives annoying warnings like unused functions to the exported functions

**Comment by overvenus** [comment]
Okay, but it still nice to have a script to provide suggestions.

**Comment by ice1000** [comment]
@overvenus Done. Can we merge this?

Ref: http://clang.llvm.org/extra/clang-tidy/

**Comment by ice1000** [comment]
Seems that clang-tidy itself can provide fixes

**Comment by ice1000** [comment]
Fixed @overvenus

**Comment by ice1000** [comment]
## Test

Applied patch:

```diff
Index: grpc-sys/grpc_wrap.cc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- grpc-sys/grpc_wrap.cc	(revision d17488a03069e8a8126e0b6ef7380d922d9fd56a)
+++ grpc-sys/grpc_wrap.cc	(date 1539453449000)
@@ -401,7 +401,7 @@
     grpc_completion_queue* cq, const char* method, size_t method_len,
     const char* host, size_t host_len, gpr_timespec deadline) {
   grpc_slice method_slice = grpc_slice_from_copied_buffer(method, method_len);
-  grpc_slice* host_slice_ptr = nullptr;
+  grpc_slice* host_slice_ptr = NULL;
   grpc_slice host_slice;
   if (host != nullptr) {
     host_slice = grpc_slice_from_copied_buffer(host, host_len);
```

Output:

```
Running clang-tidy with: -checks=clang-analyzer-* -extra-arg-before=-x -extra-arg-before=c++ -extra-arg-before=-std=c++11 -extra-arg-before=-I./grpc-sys/grpc/third_party/googletest/include -extra-arg-before=-I./grpc-sys/grpc/third_party/googletest -extra-arg-before=-I./grpc-sys/grpc/include -extra-arg-before=-I./grpc-sys/grpc/gens -extra-arg-before=-I./grpc-sys/grpc/. -extra-arg-before=-I./grpc-sys/grpc/third_party/boringssl/include -extra-arg-before=-I./grpc-sys/grpc/third_party/benchmark/include -extra-arg-before=-I./grpc-sys/grpc/third_party/zlib -extra-arg-before=-I./grpc-sys/grpc/third_party/protobuf/src -extra-arg-before=-I./grpc-sys/grpc/third_party/abseil-cpp -extra-arg-before=-I./grpc-sys/grpc/third_party/cares/cares -extra-arg-before=-I./grpc-sys/grpc/third_party/cares -extra-arg-before=-I./grpc-sys/grpc/third_party/googletest/googletest/include -extra-arg-before=-I./grpc-sys/grpc/third_party/googletest/googlemock/include ./grpc-sys/grpc_wrap.cc

Clang-Tidy stdout:
/home/ice1000/git-repos/PingCAP/grpc-rs/grpc-sys/grpc_wrap.cc:404:32: error: use nullptr [modernize-use-nullptr,-warnings-as-errors]
  grpc_slice* host_slice_ptr = NULL;
                               ^~~~~
                               nullptr

Clang-Tidy stderr:
Error while trying to load a compilation database:
Could not auto-detect compilation database for file "./grpc-sys/grpc_wrap.cc"
No compilation database found in /home/ice1000/git-repos/PingCAP/grpc-rs/grpc-sys or any parent directory
fixed-compilation-database: Error while opening fixed database: No such file or directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.
2 warnings generated.
Suppressed 1 warnings (1 in non-user code).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
1 warning treated as error
```

File: scripts/lint-grpc-sys.pl [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/perl
```
Comment (siddontang) [inline scripts/lint-grpc-sys.pl]: oh, amazing, perl !!!

**Comment by siddontang** [comment]
CI failed

**Comment by ice1000** [comment]
> CI failed

It's a problem related to #239 

```

---- ping_pong::test_secure stdout ----
testing ping pong ... thread 'ping_pong::test_secure' panicked at 'called `Result::unwrap()` on an `Err` value: RpcFinished(None)', libcore/result.rs:1009:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:71
             at libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:211
   3: std::panicking::default_hook
             at libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:476
   5: std::panicking::continue_panic_fmt
             at libstd/panicking.rs:390
   6: rust_begin_unwind
             at libstd/panicking.rs:325
   7: core::panicking::panic_fmt
             at libcore/panicking.rs:77
   8: core::result::unwrap_failed
             at libcore/macros.rs:26
   9: <core::result::Result<T, E>>::unwrap
             at libcore/result.rs:808
```

It's unrelated to this PR.

File: scripts/lint-grpc-sys.pl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+close $clangComplete or die "Failed to close $clangCompletePath: $!\n";
+
+my $completeArgs = "-checks=clang-analyzer-*$extraArgFlag@{[ join $extraArgFlag, @extraArgs ]} $grpcSysPath/grpc_wrap.cc";
+
+print "Running clang-tidy with: $completeArgs\n";
+
+no warnings 'once';
+my $pid = open3 \*WRITER, \*READER, \*ERROR, "clang-tidy-6.0 $completeArgs";
```
[the remark this answers is not in the document]
Comment (ice1000) [reply scripts/lint-grpc-sys.pl]: Now I do a search for all possible `clang-tidy` in user's Path.

File: scripts/lint-grpc-sys.pl [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+# Variable definitions (path related)
+# Put together so people can go and modify without reading the whole script
+my $grpcSysPath = './grpc-sys';
+my $grpcPath = "$grpcSysPath/grpc";
+my $clangCompletePath = "$grpcPath/.clang_complete";
+my $extraArgFlag = ' -extra-arg-before=';
+
+# Open `.clang_complete` in grpc
+open my $clangComplete, '<', $clangCompletePath or die "Failed to open $clangCompletePath: $!\n";
```
Comment (BusyJay) [inline scripts/lint-grpc-sys.pl]: Why is this necessary? Isn't just command `clang-tidy -x c++ -std=c++11` sufficient enough now that `.clang-tidy` is linked?

Comment (ice1000) [reply scripts/lint-grpc-sys.pl]: To make sure the command line argument passed to clang-tidy is consistent with grpc's own.

Comment (BusyJay) [reply scripts/lint-grpc-sys.pl]: How can they become inconsistent if not reading the files explicitly?

Comment (ice1000) [reply scripts/lint-grpc-sys.pl]: According to @overvenus , our command line arguments should be the same (and always the same in the future) as grpc's own. This Perl script is just simulating grpc's [Python script](https://github.com/pingcap/grpc/blob/eb2aa9cb3b47eb3d7980952844b32082c3e5e651/tools/distrib/run_clang_tidy.py) which runs clang-tidy. The Python script reads the file, so this script reads the file as well.

The only differences are:
+ This script works with clang-tidy-[4-6].0 as well as clang-tidy, while grpc's own only works with clang-tidy.
+ This script doesn't accept arguments itself, because we don't do further customization
+ This script only checks one file, while grpc's own checks for files specified in the arguments

Comment (BusyJay) [reply scripts/lint-grpc-sys.pl]: How can it be always the same in the future if you choose to implement a custom script? :smiley: I think it's the same as writing a command line like provided above.

**Comment by ice1000** [comment]
@overvenus @BusyJay @siddontang I think there shouldn't be anything else wrong.

**Comment by BusyJay** [comment]
Can you add this to CI check as well?

**Comment by ice1000** [comment]
@BusyJay Done, PTAL

**Comment by ice1000** [comment]
Travis CI is down.

**Comment by ice1000** [comment]
@overvenus @BusyJay @siddontang Any approving reviews?
